Tests/SonarScanner.MSBuild.Common.Test/ILoggerTests.cs

```diff
@@ -308,6 +314,40 @@ public void CLogger_SuspendAndResume()
         recorder.AssertExpectedLastOutput("error 2", ConsoleLogger.ErrorColor, true);
     }
 
+    [TestMethod]
+    public void ConsoleLogger_WriteUIWarnings_GenerateFile()
+    {
+        using var output = new OutputCaptureScope();
+        var logger = new ConsoleLogger(includeTimestamp: false, fileWrapper);
+
+        logger.LogUIWarning("uiWarn1");
+        output.AssertLastMessageEndsWith("uiWarn1"); // UI warnings should also be logged in the console
+
+        logger.LogUIWarning("uiWarn2", null);
+        output.AssertLastMessageEndsWith("uiWarn2");
+
+        logger.LogUIWarning("uiWarn3 {0}", "xxx");
+        output.AssertLastMessageEndsWith("uiWarn3 xxx");
+
+        var settings = BuildSettings.CreateNonTeamBuildSettingsForTesting(TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext));
+        logger.WriteUIWarnings(settings.SonarOutputDirectory);
+        fileWrapper.Received(1).WriteAllText(
+             Path.Combine(settings.SonarOutputDirectory, FileConstants.UIWarningsFileName),
```

Comment:
The nice part is that now we don't have to write to disk anymore.

```suggestion
        const string outputDir = "outputDir";
        logger.WriteUIWarnings(outputDir);
        fileWrapper.Received(1).WriteAllText(
             Path.Combine(outputDir, FileConstants.UIWarningsFileName),
```

Reply:
oh yeah I forgot to remove that part, good catch